custom_components/extended_openai_conversation/helpers.py

```diff
@@ -147,7 +164,13 @@ async def validate_authentication(
             api_key=api_key, base_url=base_url, organization=organization
         )
 
-    await client.models.list(timeout=10)
+    # Define an asynchronous function that lists models with a timeout using asyncio.to_thread
+    async def list_models_with_timeout():
+        # Use asyncio.to_thread to run the blocking call in a separate thread
+        return await asyncio.to_thread(client.models.list, timeout=10)
+
+    # Await the execution of the list_models_with_timeout function
+    await list_models_with_timeout()
```

Comment:
Could you change this to following?

```suggestion
    await hass.async_add_executor_job(partial(client.models.list, timeout=10))
```

Also, importing `partial` is required at the top.

```py
from functools import partial
```